frontend/src/scenes/cohorts/CohortFilters/CohortField.tsx

```diff
@@ -15,40 +16,47 @@ import {
 } from 'scenes/cohorts/CohortFilters/types'
 import { LemonDivider } from 'lib/components/LemonDivider'
 import clsx from 'clsx'
+import { resolveCohortFieldValue } from 'scenes/cohorts/cohortUtils'
+import { cohortsModel } from '~/models/cohortsModel'
 
 let uniqueMemoizedIndex = 0
 
-const useCohortLogic = (props: CohortFieldBaseProps): { logic: ReturnType<typeof cohortFieldLogic.build> } => {
+const useCohortFieldLogic = (props: CohortFieldBaseProps): { logic: ReturnType<typeof cohortFieldLogic.build> } => {
     const cohortFilterLogicKey = useMemo(
         () => props.cohortFilterLogicKey || `cohort-filter-${uniqueMemoizedIndex++}`,
         [props.cohortFilterLogicKey]
     )
     const logic = cohortFieldLogic({ ...props, cohortFilterLogicKey })
-    const { setValue } = useActions(logic)
+    const { onChange } = useActions(logic)
 
     useEffect(() => {
-        setValue(props.value)
-    }, [props.value])
+        if (props.fieldKey) {
+            onChange(props.criteria)
+        }
+    }, [resolveCohortFieldValue(props.criteria, props.fieldKey)])
```

Comment:
What's the reasoning for having this logic hook? Also, it seems like onChange does the exact checks that would happen here. `resolveCohortFieldValue`

Reply:
This keeps the logic in sync with the component props, since as of now it's impossible to do that within kea logic itself. It's hacky but we do this in `ActionFilter` too - https://github.com/PostHog/posthog/blob/812efd0b1587602de3c78b3c35f9f70e39a7897d/frontend/src/scenes/insights/ActionFilter/ActionFilter.tsx#L142-L146